ruby/ql/lib/codeql/ruby/security/XSS.qll

```diff
@@ -324,7 +324,9 @@ module StoredXss {
     OrmFieldAsSource() {
       exists(DataFlow::CallNode subSrc |
         OrmTracking::flow(subSrc, this.getReceiver()) and
-        subSrc.(OrmInstantiation).methodCallMayAccessField(this.getMethodName())
+        subSrc.(OrmInstantiation).methodCallMayAccessField(this.getMethodName()) and
+        this.getNumberOfArguments() = 0 and
+        not exists(this.getBlock())
```

Comment:
Would it help to add this instead to `OrmTracking::Config::isSink` above? Then I would hope we would prune the dataflow graph earlier.

Reply:
Ah no hang on, that doesn't make sense, because the dataflow graph stops at the receiver anyway. Ignore me.